### PR TITLE
feature: msp project for monitor

### PR DESCRIPTION
**Pull request by innerpeacez**
#### What type of this PR

Add one of the following kinds:
/kind feature

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @your-reviewer

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by snakorse — APPROVED** [review]
lgtm

File: .erda/migrations/msp/20210728-tenant.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+CREATE TABLE IF NOT EXISTS `msp_tenant`
+(
+    `id`                 varchar(100) NOT NULL COMMENT '域id',
+    `type`               varchar(10)  NOT NULL COMMENT '类型（dop 、msp）',
+    `related_project_id` bigint(20)   NOT NULL COMMENT '项目id',
+    `related_workspace`  varchar(10)  NOT NULL COMMENT '环境（ DEV、TEST、STAGING、PROD、DEFAULT）',
+    `create_time`        datetime     NOT NULL COMMENT '创建时间',
+    `update_time`        datetime     NOT NULL COMMENT '更新时间',
+    `created_at`         datetime     NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT '创建时间',
```
Comment (liuhaoyang) [inline .erda/migrations/msp/20210728-tenant.sql:9]: repeated fields

File: modules/orchestrator/dbclient/tenant.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Updated            time.Time `gorm:"column:updated" db:"updated" json:"updated" form:"updated"`
+}
+
+func (Monitor) TableName() string { return "sp_monitor" }
+
+type MonitorDB struct {
+	*gorm.DB
+}
+
+func (db *DBClient) GetMonitorByProjectIdAndWorkspace(projectID int64, workspace string) (*Monitor, error) {
```
Comment (liuhaoyang) [inline modules/orchestrator/dbclient/tenant.go:53]: Why read the monitor table in the orchestrator module?

File: modules/orchestrator/endpoints/microservice.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ import (
 
 // ListMicroServiceProject 获取使用微服务的项目列表
 func (e *Endpoints) ListMicroServiceProject(ctx context.Context, r *http.Request, vars map[string]string) (httpserver.Responser, error) {
-	_, err := user.GetUserID(r)
+	userId, err := user.GetUserID(r)
```
Comment (liuhaoyang) [inline modules/orchestrator/endpoints/microservice.go:29]: There is no need to modify the microservice project api provided by orchestrator. It should be the msp module that calls this api, and then aggregates tenant data and old data

File: modules/orchestrator/services/tenant/tenant.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+type Tenant struct {
+	db *dbclient.DBClient
+	hc *httpclient.HTTPClient
+}
+
+// Option 应用实例对象配置选项
+type Option func(*Tenant)
+
+// New 新建应用实例 service
+func New(options ...Option) *Tenant {
```
Comment (liuhaoyang) [inline modules/orchestrator/services/tenant/tenant.go:36]: Operations on msp tenants should be done inside the msp module

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

File: modules/msp/instance/db/monitor/monitor.go [hunk 5]
```diff
@@ -65,6 +65,30 @@ type CompatibleTerminusKey struct {
 	TerminusKeyRuntime string `gorm:"column:terminus_key_runtime"`
 }
 
+func (db *MonitorDB) GetMonitorByProjectId(projectID int64) ([]*Monitor, error) {
+	var monitors []*Monitor
+	err := db.Where("`project_id` = ?", projectID).Find(&monitors).Error
```
Comment (liuhaoyang) [inline modules/msp/instance/db/monitor/monitor.go:70]: id_delete = '0' query condition needs to be added

File: modules/msp/instance/db/monitor/monitor.go [hunk 6]
```diff
@@ -65,6 +65,30 @@ type CompatibleTerminusKey struct {
...
+	}
+	if err != nil {
+		return nil, err
+	}
+	return monitors, nil
+}
+
+func (db *MonitorDB) GetMonitorByProjectIdAndWorkspace(projectID int64, workspace string) (*Monitor, error) {
+	monitor := Monitor{}
+	err := db.Where("`project_id` = ?", projectID).Where("`workspace` = ?", workspace).Find(&monitor).Error
```
Comment (liuhaoyang) [inline modules/msp/instance/db/monitor/monitor.go:82]: id_delete = '0' query condition needs to be added

File: modules/msp/tenant/project/project.service.go [hunk 7]
```diff
@@ -0,0 +1,245 @@
...
+	var projects []*pb.Project
+
+	// request orch for history project
+	params := url.Values{}
+	for _, id := range req.ProjectId {
+		params.Add("projectId", strconv.FormatInt(id, 10))
+	}
+	userID := apis.GetUserID(ctx)
+	orgID := apis.GetOrgID(ctx)
+	do := httpclient.New().Get(discover.Orchestrator()).
```
Comment (liuhaoyang) [inline modules/msp/tenant/project/project.service.go:95]: Use bundle instead of httpclient directly

Comment (liuhaoyang) [reply modules/msp/tenant/project/project.service.go:95]: Each call creates an new httpclient, which will cause `http.Transport` leakage

File: modules/msp/tenant/project/provider.go [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+package project
+
+import (
+	"github.com/jinzhu/gorm"
+
+	logs "github.com/erda-project/erda-infra/base/logs"
+	servicehub "github.com/erda-project/erda-infra/base/servicehub"
+	transport "github.com/erda-project/erda-infra/pkg/transport"
```
Comment (liuhaoyang) [inline modules/msp/tenant/project/provider.go:21]: you do not need to explicitly declare aliases

File: modules/openapi/api/apis/msp/tenant/project/get_msp_projects.go [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+	Doc:         "GET MSP projects",
+}
+
+func attachProjectParams(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("User-ID")
+	orgID := r.Header.Get("Org-ID")
+
+	// get permissions
+	client := httpclient.New()
+	var perms apistructs.ScopeRoleList
```
Comment (liuhaoyang) [inline modules/openapi/api/apis/msp/tenant/project/get_msp_projects.go:46]: Use bundle instead of httpclient directly

**Review by liuhaoyang — CHANGES_REQUESTED** [review]

File: modules/msp/tenant/project/provider.go (hunk 8 above)
Comment (innerpeacez) [reply modules/msp/tenant/project/provider.go:21]: automatically generated code

**Review by liuhaoyang — APPROVED** [review]

**Pull request merged**